pkgs/by-name/li/linien-gui/package.nix

```diff
@@ -4,16 +4,44 @@
   qt5,
 }:
 
-python3.pkgs.buildPythonApplication rec {
+let
+  loosenNumpy =
+    p: pattern:
+    p.overridePythonAttrs (old: {
+      postPatch =
+        (old.postPatch or "")
+        # Extra empty line is here due to potential last line of existing
+        # postPatch ending with `\`.
+        + ''
+
+          substituteInPlace pyproject.toml \
+            --replace-fail "${pattern}" "numpy"
+        '';
+    });
+  python = python3.override {
+    packageOverrides = self: super: {
+      numpy = super.numpy_1;
+      # NOTE: Both of these packages in principal support building against
+      # Numpy 2 and using with Numpy 1, but unfortunately we still have to
+      # rebuild them using Numpy 1 because we don't have yet good separation of
+      # runtime v.s build time depdencies in our Python ecosystem. See:
+      #
+      # https://discourse.nixos.org/t/several-comments-about-priorities-and-new-policies-in-the-python-ecosystem/51790
+      pandas = loosenNumpy super.pandas "numpy>=2.0";
+      scipy = loosenNumpy super.scipy "numpy>=2.0.0,<2.5";
```

Comment:
Doesn't it make sense to do this in the pandas and scipy packages? That way everybody can override numpy more easily.

Reply:
If you prefer we can do this in a separate PR. But I definitely think it should be done in those packages because the override might otherwise break with the next update.

Reply:
> If you prefer we can do this in a separate PR. But I definitely think it should be done in those packages because the override might otherwise break with the next update.

I prefer doing it all in this - single PR.